server/src/main/java/org/opensearch/cluster/routing/IndexShardRoutingTable.java

```diff
@@ -292,6 +292,77 @@ public ShardIterator activeInitializingShardsRankedIt(
         return new PlainShardIterator(shardId, ordered);
     }
 
+    /**
+     * Returns an iterator over active and initializing shards, shards are ordered by weighted round-robin scheduling
+     * policy.
+     * @param wrrWeight Weighted round-robin weight entity
+     * @param nodes discovered nodes in the cluster
+     * @return an iterator over active and initializing shards, ordered by weighted round-robin
+     * scheduling policy. Making sure that initializing shards are the last to iterate through.
+     */
+    public ShardIterator activeInitializingShardsWRR(WRRWeights wrrWeight, DiscoveryNodes nodes, WRRShardsCache cache) {
+        final int seed = shuffler.nextSeed();
+        List<ShardRouting> ordered = new ArrayList<>(activeShards.size() + allInitializingShards.size());
```

Comment:
yeah thats true. But we can start with this initial capacity since we can expect minimum of these many shards and list can resize if more shards are added